mmocr/datasets/preparers/obtainers/aws_s3_obtainer.py

```diff
@@ -0,0 +1,237 @@
+
+        extracted = False
+        if osp.exists(dst_path):
+            name = set(os.listdir(dst_path))
+            if '.finish' in name:
+                extracted = True
+            elif '.finish' not in name and len(name) > 0:
+                while True:
+                    c = input(f'{dst_path} already exists when extracting '
+                              '{zip_name}, unzip again? (y/N) ') or 'N'
+                    if c.lower() in ['y', 'n']:
+                        extracted = c == 'n'
+                        break
+        if extracted:
+            open(osp.join(dst_path, '.finish'), 'w').close()
+            print(f'{zip_name} has been extracted. Skip')
+            return
+        mkdir_or_exist(dst_path)
+        print(f'Extracting: {osp.basename(src_path)}')
+        if src_path.endswith('.zip'):
+            try:
+                import zipfile
+            except ImportError:
+                raise ImportError(
+                    'Please install zipfile by running "pip install zipfile".')
+            with zipfile.ZipFile(src_path, 'r') as zip_ref:
+                zip_ref.extractall(dst_path)
+        elif src_path.endswith('.tar.gz') or src_path.endswith(
+                '.tar') or src_path.endswith('.tgz'):
+            if src_path.endswith('.tar.gz'):
+                mode = 'r:gz'
+            elif src_path.endswith('.tar'):
+                mode = 'r:'
+            elif src_path.endswith('tgz'):
+                mode = 'r:gz'
+            try:
+                import tarfile
+            except ImportError:
+                raise ImportError(
+                    'Please install tarfile by running "pip install tarfile".')
+            with tarfile.open(src_path, mode) as tar_ref:
+                tar_ref.extractall(dst_path)
+
+        open(osp.join(dst_path, '.finish'), 'w').close()
+        if delete:
+            os.remove(src_path)
+
+    def move(self, mapping: List[Tuple[str, str]]) -> None:
+        """Rename and move dataset files one by one.
+
+        Args:
+            mapping (List[Tuple[str, str]]): A list of tuples, each
+            tuple contains the source file name and the destination file name.
+        """
+        for src, dst in mapping:
+            src = osp.join(self.data_root, src)
+            dst = osp.join(self.data_root, dst)
+
+            if '*' in src:
+                mkdir_or_exist(dst)
+                for f in glob.glob(src):
+                    if not osp.exists(
+                            osp.join(dst, osp.relpath(f, self.data_root))):
+                        shutil.move(f, dst)
+
+            elif osp.exists(src) and not osp.exists(dst):
+                mkdir_or_exist(osp.dirname(dst))
+                shutil.move(src, dst)
+
+    def clean(self) -> None:
+        """Remove empty dirs."""
+        for root, dirs, files in os.walk(self.data_root, topdown=False):
+            if not files and not dirs:
+                os.rmdir(root)
```

Comment:
Duplicated code can be avoided by inheriting this class from `NaiveDataObtainer`